### ISSUE TITLE
New Rule: Lock objects should be read-only

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#5542]
As proposed by @Corniel here: https://community.sonarsource.com/t/new-rule-for-c-and-vb-net-locks-should-be-released/60951/2

### Non-compliant
```
private object lock = new object(); // Noncompliant {{lock objects should be read-only.}}
```

### Compliant
```
private readonly object lock = new object();
```

**Comment by pavel-mikula-sonarsource** [comment]
To distinguish between S2933 and this: This should trigger on fields used by `lock` and `SyncLock` statements.

**Comment by antonioaversa** [comment]
Seems to be a duplicate of https://github.com/SonarSource/sonar-dotnet/issues/6701, implementation of https://sonarsource.github.io/rspec/#/rspec/S2445/csharp for C#
